### ISSUE TITLE
Clarify mutability of  AggregationTemporality and monotonicity 

### ISSUE THREAD

**Issue by cevian** [open-telemetry/opentelemetry-specification#2250]
**What are you trying to achieve?**

The [metric spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#opentelemetry-protocol-data-model) says that 

```
Note: The same Resource, name and Attributes but differing point kind coming out of an OpenTelemetry SDK is considered an "error state" that SHOULD be handled by an SDK.
```

I'd like some clarification about fields like `aggregation_temporality` and `is_monotonic` are those fields allowed to change values with the same "stream" (same Resource, name and Attributes) or is there some guarantee on consistency there as well. 

In my mind it's very strange if a stream changes either of those fields in the middle, but maybe there are some use cases that require this?

**Comment by jmacd** [comment]
I fixed this, to some degree, in #2317.
Neither should change mid-stream, however duplicate instrument registration conflicts could lead to a mixture of values for the what looks like the same metric name.

**Comment by bogdandrutu** [comment]
@jmacd @cevian there are scenarios where this may happen, in cases where a restart happens to the app. A restart does not necessary mean a change in the Resource/Attribute etc. There are ways to detect stream restart like the start_time in the points, or a previous staleness point.
